woocommerce/woocommerce
# Automatically Deactivate Merged Packages

Once a feature plugin has been merged it is no longer necessary for it to be activated. In some cases, having it activated can actually lead to problems. This will automatically disable any packages that we have marked as "merged".

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

One of the caveats to the version of WooCommerce Blocks in the wild is that all of its class files are registered with the Jetpack Autoloader using a higher version that WooCommerce Core. This means that it will preferentially load classes from the active version of Blocks instead of WooCommerce Core. This will continue to be the case for several years until WooCommerce Core has a higher version than the current version of WooCommerce Blocks.

As such, this pull request adds support for automatically disabling WooCommerce Blocks when it is detected as active. I migrated the existing auto-disabling that was in place for WooCommerce Admin since this is essentially the same thing.

Closes #41557.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install WooCommerce Blocks & [WooCommerce Admin](https://github.com/woocommerce/woocommerce-admin/releases/download/v3.3.2-plugin/woocommerce-admin.zip)
2. Make sure that the plugins are automatically deactivated on activation.
3. Set the `WC_ALLOW_MERGED_FEATURE_PLUGINS` constant to `true`. When you do this, it'll disable the automatic deactivation functionality.
4. Run `composer require woocommerce/woocommerce-blocks 11.6.0` from inside `plugins/woocommerce`. This will downgrade the version in WooCommerce. Normally this would mean that the files aren't used, but in the case of this PR, it means that they'll be used instead of the new version of files within the feature plugin. The Jetpack Autoloader should take care of this automatically.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Automatically disable feature plugins that have been merged into WooCommerce Core.

#### Comment 

</details>
